src/app/[bookId]/components/book-comments-form.tsx

```diff
@@ -0,0 +1,71 @@
+'use client'
+
+import React from 'react'
+import { useForm, Controller } from 'react-hook-form'
+
+import BooksAPI from '@/api/books-api'
+import { Button } from '@/components/ui/button'
+import { Textarea } from '@/components/ui/textarea'
+import useUserStore from '@/hooks/store/use-user-store'
+
+interface BookCommentsFormProps {
+  bookId: string
+}
+
+const BookCommentsForm = ({
+  bookId,
+  updateBook,
+}: BookCommentsFormProps & { updateBook: () => void }) => {
+  const { user } = useUserStore()
+
+  const {
+    handleSubmit,
+    control,
+    watch,
+    formState: { isSubmitting },
+  } = useForm({
+    defaultValues: { content: '' },
+  })
+
+  const allFields = watch()
+  const allFieldsFilled = allFields.content.trim() !== ''
+
+  const onSubmit = async (data: { content: string }) => {
+    if (!user) return
+    const comment = { userId: user._id, comment: data.content }
```

Comment:
```suggestion
    const comment = { userId: user!._id, comment: data.content }
```

there is no need to check, if there is no user, there will be no button with submit

Reply:
Fixed it